examples/Example21/gammas.cuh

```diff
@@ -0,0 +1,255 @@
+// SPDX-FileCopyrightText: 2022 CERN
+// SPDX-License-Identifier: Apache-2.0
+
+#include "AdeptIntegration.cuh"
+
+#include <AdePT/BVHNavigator.h>
+
+#include <CopCore/PhysicalConstants.h>
+
+#include <G4HepEmGammaManager.hh>
+#include <G4HepEmGammaTrack.hh>
+#include <G4HepEmTrack.hh>
+#include <G4HepEmGammaInteractionCompton.hh>
+#include <G4HepEmGammaInteractionConversion.hh>
+#include <G4HepEmGammaInteractionPhotoelectric.hh>
+// Pull in implementation.
+#include <G4HepEmGammaManager.icc>
+#include <G4HepEmGammaInteractionCompton.icc>
+#include <G4HepEmGammaInteractionConversion.icc>
+#include <G4HepEmGammaInteractionPhotoelectric.icc>
+
+template <typename Scoring>
+__global__ void TransportGammas(adept::TrackManager<Track> *gammas, Secondaries secondaries,
+                                MParrayTracks *leakedQueue, Scoring *userScoring)
+{
+  using VolAuxData = AdeptIntegration::VolAuxData;
+#ifdef VECGEOM_FLOAT_PRECISION
+  const Precision kPush = 10 * vecgeom::kTolerance;
+#else
+  const Precision kPush = 0.;
+#endif
+  constexpr Precision kPushOutRegion = 10 * vecgeom::kTolerance;
+  constexpr int Pdg                  = 22;
+  int activeSize                     = gammas->fActiveTracks->size();
+  for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < activeSize; i += blockDim.x * gridDim.x) {
+    const int slot      = (*gammas->fActiveTracks)[i];
+    Track &currentTrack = (*gammas)[slot];
+    auto energy         = currentTrack.energy;
+    auto pos            = currentTrack.pos;
+    auto dir            = currentTrack.dir;
+    auto navState       = currentTrack.navState;
+    const auto volume   = navState.Top();
+    adeptint::TrackData trackdata;
+    // the MCC vector is indexed by the logical volume id
+    int lvolID                = volume->GetLogicalVolume()->id();
+    VolAuxData const &auxData = userScoring->GetAuxData_dev(lvolID);
+    assert(auxData.fGPUregion > 0); // make sure we don't get inconsistent region here
```

Comment:
The same, remove, it might be that we want it at some point.

Reply:
removed